packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -120,6 +170,57 @@ export default class SaytDriverPlugin implements Plugin {
     return [searchTerms];
   }
 
+  /**
+   * Extracts query and products from the given response.
+   *
+   * @param response An object containing the original query and product records.
+   * @returns An object containing the query and an array of valid simplified products.
+   */
+  searchCallback(response: Results): ProductsResponseSection {
+    let filter;
+    const { query, records } = response;
+    const mappedRecords = records.map(record => {
+      try {
+        filter = this.parseRecord(record);
+      } catch(error) {
+        return;
+      }
+      const { data, firstVariant, nonvisualVariants } = filter;
+
+      return {
+        title: data.title,
+        price: nonvisualVariants[0].originalPrice,
+        imageSrc: firstVariant.productImage,
+        imageAlt: data.title,
+        productUrl: firstVariant.productImage,
+        // @TODO Handle variants
+      }
+    }).filter(Boolean);
+    return {
```

Comment:
I guess it's fine for this case, but I want to be careful about doing too much processing and stripping down the data too much without giving access to the full response.

Reply:
Yeah, I'm unsure of whether or not we want to filter anything at all. In the meantime, this function works for the apparel data and will be updated to be more complete (even before adding a proper client-friendly transformer).